posts/2023-07-05-select-region-shapefile.ipynb

```diff
@@ -0,0 +1,2641 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "df['mean'].plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c26cd8a-2c78-46f3-a743-f5e8f1449b58",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Creating a mask with rasterio"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9103fd60-1cc8-4add-a026-8ef2921e0362",
+   "metadata": {},
+   "source": [
+    "An alternative way is to create a mask for the region as a xarray DataArray with the same shape as the data. We can then use to this to select the region.<br>\n",
+    "We start by opening the shapefile with GeoPandas. GeoPandas combines the capabilities of pandas and shapely, providing geospatial operations in pandas."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "8973546c-25d6-4142-a4c5-4f72ad53b8b9",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0    MULTIPOLYGON (((149.97620 -37.50516, 149.97620...\n",
+       "Name: geometry, dtype: geometry"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "gdf = gpd.read_file(shapefile)\n",
+    "gdf.geometry"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e65f3927-9b6c-439a-8147-18e2061bf149",
+   "metadata": {},
+   "source": [
+    "There is no `crs` (Coordinate reference System) defined for this shapefile so I'm using the array `crs` which is a standard one. If the `crs` for the shapefile and the data array are different you have to reproject the shapefile."
```

Comment:
I would change
"reproject the shapefile" 
into
"change the projection of the shapefile"

Reply:
thanks I've followed the suggestion